src/ccmain/thresholder.cpp

```diff
@@ -190,11 +190,6 @@ std::tuple<bool, Image, Image, Image> ImageThresholder::Threshold(
   Image pix_binary = nullptr;
   Image pix_thresholds = nullptr;
 
-  if (image_width_ > INT16_MAX || image_height_ > INT16_MAX) {
```

Comment:
1) As I said elsewhere, this block was copied from another function in the same file. You should delete the other block too.

Reply:
There might be security issues in allowing unlimited file size.

You should check if Leptonica has any image size limits.